### DISCUSSION TITLE
How can I pass custom information in DataFetching enviroment

### DISCUSSION THREAD

**Discussion by vyj7** [graphql-java/graphql-java#2205] [Q&A] [answered]
I used to pass values to environment by 
`enviroment.getVariables.put("somObject",obj);`
 The DataFetching environment has been made as ImmutableMapWithNullValues<String, Object> so the above method will fail.
Is there a way where I can add custom objects to the Datafetching environment?

This is how I am currently doing it:

` 
 Map variableMap = Maps.newHashMap(env.getVariables());    
variableMap.put("somObject",Obj);
 environment  =  DataFetchingEnvironmentImpl.newDataFetchingEnvironment(env).variables(variableMap).build();`

**Comment by bbakerman** [comment, marked as answer]
Are you somehow passing this mutable state down to child data fetchers - is that your use case?

See https://www.graphql-java.com/blog/deep-dive-data-fetcher-results/ and the section on `graphql.execution.DataFetcherResult`

This is how you pass context from one DataFetcher to a lower one.

If you dont have a parent -> chidl relationship in the your DataFecthers then local context wont work.

but you can create a mutable map in your global context object `graphql.ExecutionInput#getContext` and whack what ever you like in that.

You know shared mutable state is bad right - especially in multi threaded systems.  A mutable map of 'stuff" will probably bite you.
